src/rviz/visualization_manager.cpp

```diff
@@ -487,62 +512,134 @@ void VisualizationManager::onUpdate()
-      QString filename;
-      filename.sprintf("%s/%08d.jpg", dump_images_config_->folder.c_str(), uint(++dumped_frame_count_));
+      if (screenshot_.width() % 2 || screenshot_.height() % 2) {
+          // ensure screenshot width are height are divisible by two
+          int w = screenshot_.width();
+          int h = screenshot_.height();
+          screenshot_ = screenshot_.copy(0, 0, w - (w%2), h - (h%2));
+      }
+      QImage img = screenshot_.toImage();
 
-      QImageWriter writer(filename);
-      if (!writer.write(screenshot_.toImage()))
-      {
-        QString error_message;
-        if (writer.error() == QImageWriter::UnsupportedFormatError)
-        {
-          QString suffix = filename.section('.', -1);
-          QString formats_string;
-          QList<QByteArray> formats = QImageWriter::supportedImageFormats();
-          formats_string = formats[0];
-          for (int i = 1; i < formats.size(); i++)
+      if ( private_->venc_ == NULL){
+          private_->venc_width_ = screenshot_.width();
+          private_->venc_height_ = screenshot_.height();
+
+          // RVN:FIXME implement more formats if needed
+          if ( img.format() != QImage::Format_RGB32 ){
+              ROS_ERROR("Detected unsupported screenshot image format: %u",img.format());
+              exit(EXIT_FAILURE);
+          }
+
+          ROS_INFO("Init rviz video encoder: %dx%d @ %d/%d fps, input format %d",private_->venc_width_,private_->venc_height_,
+                   dump_images_config_->fpsNum, dump_images_config_->fpsDen, img.format());
+          // this is the first frame- initialize encoders
+          VideoEncodeParams params = {0};
+
+          params.width = private_->venc_width_;
+          params.height= private_->venc_height_;
+          params.fpsNum = dump_images_config_->fpsNum;
+          params.fpsDen = dump_images_config_->fpsDen;
+
+          // RVN:TODO expose as CLI parameters
+          params.crf = RVN_LIBAV_CRF;
+          params.preset = RVN_LIBAV_PRESET;
+          params.maxWidth = RVN_LIBAV_MAXWIDTH;
+
+          params.output_path = dump_images_config_->captured_path.c_str();
+          private_->venc_ = video_encoder_init(params);
+          if (private_->venc_ == NULL){
+              ROS_ERROR("Failed to create capture video encoder");
+              exit(EXIT_FAILURE);
+          }
+
+          params.keyed = 1;
+          params.output_path = dump_images_config_->keyed_path.c_str();
+          private_->venc_keyed_ = video_encoder_init(params);
+          if (private_->venc_keyed_ == NULL){
+              ROS_ERROR("Failed to create capture keyed video encoder");
+              exit(EXIT_FAILURE);
+          }
+
+      }else{
+
+          // Ensure image dimensions have not changed
+          // RVN:TODO perhaps support this (if we have a use case)
+          if (screenshot_.width() != private_->venc_width_ || screenshot_.height() != private_->venc_height_)
```

Comment:
This is currently being triggered:
`Detected change in screenshot dimensions: was -1493129544x1866242144, now is 1881x2088`
This is a result of venc_ not being initialized.